### ISSUE TITLE
Request: Cart in header

### ISSUE THREAD

**Issue by wiztechinc** [zencart/zencart#2951]
I always give my customers this option and use it on dang near every site I ever work on and many templates also use it. As a user, I always need that reminder/guide (no, I don't claim to know about other users). But this is such an easy add, I don't see why we don't have it in the default. I have it coded for (simple) responsive too but if I do this, I may need assistance in the default responsive because I don't work in it and don't want to unless something holds a threat of death over my head. I can give you the url to my demo site so you can see what I do if necessary.

**Comment by drbyte** [comment]
Yes, a demo link would be useful, and perhaps also a screenshot for quick-reference pasted here in this Issue.

Also useful would be the code needed, and in which files, to make it work. (I'm not bothered much about the markup used by whatever template you're using; more interested in the PHP code required for how you've chosen to implement it.) 

Both of those will help with inspiration towards an implementation.

Storeowners often want things displayed differently from the default, but I agree it's helpful to have a starting point for them to work from.

**Comment by wiztechinc** [comment]
the demo is https://deliatest.com/readytogo

It's inserted like the search header box. I don't even know where I pulled this from originally because I don't see the plugin download zip (I have them dating back to 2006.)
![image](https://user-images.githubusercontent.com/5367263/70377559-e1040000-18e3-11ea-93eb-3ea69ade58bc.png)
code for template file:
<?php
/**
 * Side Box Template
 *
 * @package templateSystem
 * @copyright Copyright 2003-2006 Zen Cart Development Team
 * @copyright Portions Copyright 2003 osCommerce
 * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
 * @version $Id: tpl_search_header.php 9755 2008-09-19 19:47:22Z ajeh $
 */
  $content = "";
   $content .= '<div id="shoppingCartBox"><ul><li class="first"><a href="'. zen_href_link(FILENAME_SHOPPING_CART, '', 'NONSSL').'">'. $_SESSION['cart']->count_contents().' item(s) '. $currencies->format($_SESSION['cart']->show_total()).'</a>';
   $content .= '</li><li class="last">';
   if (($_SESSION['customer_id'])) 
   { 
    $content .= '<a href="'. zen_href_link(FILENAME_LOGOFF, '', 'SSL').'">'.HEADER_TITLE_LOGOFF. '</a> | ';
      } elseif (STORE_STATUS == '0' && (!$_SESSION['COWOA'])) {
    $content .='<a href="'.zen_href_link(FILENAME_LOGIN, '', 'SSL').'">'.HEADER_TITLE_LOGIN. '</a> | ';
 }
  $content .= '<a href="'.zen_href_link(FILENAME_ACCOUNT, '', 'SSL').'">'.HEADER_TITLE_MY_ACCOUNT.'</a>';
   if ($_SESSION['cart']->count_contents() != 0) { 
  $content .= '<a href="'. zen_href_link(FILENAME_CHECKOUT_SHIPPING, '', 'SSL'). '"><span class="checkOut">&#10148; '.  HEADER_TITLE_CHECKOUT. '</span></a>';
   }
   $content .= '</li></ul></div>';
?>

**Comment by lat9** [comment]
I like the looks of that change, too, but not all stores will (some use jQuery add-to-cart that will get 'in conflict').  I'm all for the change, so long as there's a way to enable/disable without carving up code.

**Comment by lat9** [comment]
Remember, too, to use `zen_is_logged_in()` instead of `$_SESSION['customer_id']` and `zen_in_guest_checkout()` instead of `$_SESSION['COWOA']`.

**Comment by wiztechinc** [comment]
Thanks lat9. Once I started digging into code, realized there were changes I just didn't realized was there like the guest checkout (yay!). But I had removed it by the time I realized it was there! Sigh.

I set up the classic template first and it looks like this 
![image](https://user-images.githubusercontent.com/5367263/70474594-e78ba680-1aa0-11ea-94aa-dbf99d461910.png)

I  pulled the navMain div out of the tpl_header and made it into a sidebox as I'm aiming for making this easier on others to move this navMain around. There's no jquery as it's really just a rework of the way it's always been done, more along the line of display and not functionality - adding in the items/dollars and making that  and the my account visible at all times. (No one should ever have to wonder where that my account link is.)

So in reality this isn't much of change but it's a change to a very long standing bit of zen cart default that I have never liked.

I have created several mods for the header in order to be able to plunk in code such as the manufacturer dropdown. All with the aim of making this even more modular than it is now. So I got more. I just don't want to upload more plugins that I have to be responsible for due to past health issues and the real fact I don't know I can maintain anything on a long term basis. 

My primary focus in zen cart work is usability - adding items, reworking items to be more sales- oriented, improve the user experience and to ease use of the cart itself for the owner.

As most folks working on this are more coders than user experience, I figure this is the best way for me to contribute and then also makes my mods unnecessary later (I love that part).
